libraries/freertos_plus/standard/freertos_plus_tcp/source/FreeRTOS_DNS.c

```diff
@@ -1590,29 +1609,37 @@ static Socket_t prvCreateDNSSocket( void )
 {
 Socket_t xSocket;
 struct freertos_sockaddr xAddress;
-BaseType_t xReturn;
+BaseType_t xReturn, xResult = pdPASS;
 
 	/* This must be the first time this function has been called.  Create
 	the socket. */
 	xSocket = FreeRTOS_socket( FREERTOS_AF_INET, FREERTOS_SOCK_DGRAM, FREERTOS_IPPROTO_UDP );
 	if( ( xSocket == FREERTOS_INVALID_SOCKET ) || ( xSocket == NULL ) )
 	{
-		return NULL;
+		xResult = pdFAIL;
 	}
 
```

Comment:
Or maybe like this:
~~~c
    if( ( xSocket == FREERTOS_INVALID_SOCKET ) || ( xSocket == NULL ) )
    {
        xResult = pdFAIL;
        /* There was an error, return NULL. */
        xSocket = NULL;
    }
    else
    {
        /* Auto bind the port. */
        xAddress.sin_port = 0U;
        xReturn = FreeRTOS_bind( xSocket, &xAddress, sizeof( xAddress ) );
~~~

Reply:
Done!